sqlglot/transforms.py

```diff
@@ -628,11 +628,33 @@ def eliminate_full_outer_join(expression: exp.Expression) -> exp.Expression:
             expression_copy = expression.copy()
             expression.set("limit", None)
             index, full_outer_join = full_outer_joins[0]
+
+            tables = (
+                full_outer_join.find(exp.Table).alias_or_name,
+                expression.args["from"].alias_or_name,
+            )
+            join_conditions = full_outer_join.args.get("on") or exp.And(
+                expressions=[
+                    exp.EQ(
+                        this=exp.Column(this=col, table=tables[0]),
+                        expression=exp.Column(this=col, table=tables[1]),
+                    )
+                    for col in full_outer_join.args.get("using")
+                ]
+            )
```

Comment:
This AST is incorrect, there's no `expressions` child in `And`. You could use the `and_` helper instead:

```python
and_(
    *[
        exp.column(col, tables[0].eq(exp.column(col, tables[1]))
        for col in full_outer_join.args.get("using")
    ]
)
```